invoices/invoiceregistry.go

```diff
@@ -1027,6 +1032,53 @@ func (i *InvoiceRegistry) notifyExitHopHtlcLocked(
 	)
 
 	callback := func(inv *Invoice) (*InvoiceUpdateDesc, error) {
+		// Provide the invoice to the settlement interceptor to allow
+		// the interceptor's client an opportunity to manipulate the
+		// settlement process.
+		clientReq := HtlcModifyRequest{
+			ExitHtlcCircuitKey: ctx.circuitKey,
+			ExitHtlcAmt:        ctx.amtPaid,
+			ExitHtlcExpiry:     ctx.expiry,
+			CurrentHeight:      uint32(ctx.currentHeight),
+			Invoice:            *inv,
+		}
+		interceptSession := i.cfg.HtlcModifier.Intercept(
+			clientReq,
+		)
+
+		// If the interceptor service has provided a response, we'll
+		// use the interceptor session to wait for the client to respond
+		// with a settlement resolution.
+		var interceptErr error
+		interceptSession.WhenSome(func(session InterceptSession) {
+			log.Debug("Waiting for client response from invoice " +
```

Comment:
can we not hide pretty much all of this within the `HTLCModifier`'s `Intercept call instead ? 
Since I think this will look roughly the same no matter where this is being called from. ie, i think this can then just be: 

```
               err := i.cfg.HtlcModifier.Intercept(
			clientReq, func(resp HtlcModifyResponse) func() {
				if resp.AmountPaid != 0 {
					ctx.amtPaid = resp.AmountPaid
				}
			}
		)
		if err != nil {
			return nil, fmt.Errorf("error during invoice HTLC "+
				"interception: %w", err)
		}
```

Reply:
the various session channels can then be unexported and the waiting no them can happen within `Intercept` itself

Reply:
I like that very much! Was able to simplify things quite a bit.